mllib/src/main/scala/org/apache/spark/ml/classification/LogisticRegression.scala

```diff
@@ -595,55 +831,104 @@ class LogisticRegressionModel private[spark] (
    * Predict label for the given feature vector.
    * The behavior of this can be adjusted using [[thresholds]].
    */
-  override protected def predict(features: Vector): Double = {
+  override protected def predict(features: Vector): Double = if (isMultinomial) {
+    super.predict(features)
```

Comment:
Would you mind elaborating? This calls ends up calling `predictRaw(features).argmax`, which equates to `margins(features).argmax`. What specialized version are you referring to? Thanks.